From d5fabc438689d910aaeba7ca5d000150fcc45f99 Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Wed, 22 Jul 2020 13:22:08 +0700
Subject: [PATCH 259/439] ARM: dts: r8a7744: Add VCP/VPC nodes

Add device nodes of Video Processing Unit Cache (VPC IP) and
Multi-codec module (VCP IP) to the R8A7744 (RZ/G1N) SoC dtsi.

These IPs are controlled by Renesas UVCS kernel driver module
(uvcs-kernel-module).

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7744.dtsi | 42 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7744.dtsi b/arch/arm/boot/dts/r8a7744.dtsi
index 4f1e05c..d0e580c 100644
--- a/arch/arm/boot/dts/r8a7744.dtsi
+++ b/arch/arm/boot/dts/r8a7744.dtsi
@@ -1123,6 +1123,48 @@
 			status = "disabled";
 		};
 
+		vcp0: vcp0@fe900000 {
+			compatible = "renesas,vcp0";
+			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+			interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+				     <0 259 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7744_CLK_VCP0>;
+			power-domains = <&sysc R8A7744_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0: vcp0@fe908000 {
+			compatible = "renesas,vpc0";
+			reg = <0 0xfe908000 0 0x90>;
+			clocks = <&mstp1_clks R8A7744_CLK_VPC0>;
+			power-domains = <&sysc R8A7744_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1: vcp0@fe918000 {
+			compatible = "renesas,vpc1";
+			reg = <0 0xfe918000 0 0x90>;
+			/* clocks = <&mstp1_clks R8A7744_CLK_VPC1>; */
+			power-domains = <&sysc R8A7744_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0xy: vcp0@fe960380 {
+			compatible = "renesas,vpc0xy";
+			reg = <0 0xfe960380 0 0x4>;
+			/* clocks = <&mstp1_clks R8A7744_CLK_VPC0>; */
+			power-domains = <&sysc R8A7744_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1xy: vcp0@fe960384 {
+			compatible = "renesas,vpc1xy";
+			reg = <0 0xfe960384 0 0x4>;
+			/* #clocks = <&mstp1_clks R8A7744_CLK_VPC1>; */
+			power-domains = <&sysc R8A7744_PD_ALWAYS_ON>;
+			renesas,#ch = <0>;
+		};
+
 		rcar_sound: sound@ec500000 {
 			/*
 			 * #sound-dai-cells is required
-- 
2.7.4

